src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,338 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+
+//
+// This captures information of how various key methods of building with MSBuild ran.
+//
+namespace Microsoft.Build.Eventing
+{
+    // /OnlyProviders=*MSBuild-Profiling
+    [EventSource(Name = "MSBuild-Profiling")]
+    public sealed class MSBuildEventSource : EventSource
+    {
+        #region Singleton instance
+
+        // define the singleton instance of the event source
+        public static MSBuildEventSource Log = new MSBuildEventSource();
+
+        private MSBuildEventSource() { }
+
+        #endregion
+
+        #region Events and NonEvents
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to collecting a set of items, mutating them in a specified way, and saving the results.
+        /// </summary>
+        public void ApplyStart()
+        {
+            if (IsEnabled()) WriteEvent(1);
```

Comment:
You should remove the IsEnabled calls from all of the event methods.  IsEnabled is called inside of WriteEvent so this is just duplicate work.

If there is something that is an argument that is expensive to compute, you should put that code in an `IsEnabled` check, but that should occur outside of `Event` methods.  If you want it to happen in a method that is inside of the `EventSource` class, you can create a method and mark it as `[NonEvent]` so that it's not treated as an event.